manticore/manticore.py

```diff
@@ -27,8 +27,69 @@
 
 logger = logging.getLogger('MANTICORE')
 
+def loggerSetState(logger, stateid):
+    logger.filters[0].stateid = stateid
 
-def makeDecree(program, concrete_data=''):
+class ContextFilter(logging.Filter):
+    '''
+    This is a filter which injects contextual information into the log.
+    '''
+    def filter(self, record):
+        if hasattr(self, 'stateid') and isinstance(self.stateid, int):
+            record.stateid = '[%d]' % self.stateid
+        else:
+            record.stateid = ''
+        return True
+
+ctxfilter = ContextFilter()
+
+logging.basicConfig(format='%(asctime)s: [%(process)d]%(stateid)s %(name)s:%(levelname)s: %(message)s', stream=sys.stdout, level=logging.ERROR)
+loggers = ['MANTICORE', 'VISITOR', 'EXECUTOR', 'CPU', 'REGISTERS', 'SMT', 'MEMORY', 'PLATFORM']
+for loggername in loggers:
+    logging.getLogger(loggername).addFilter(ctxfilter)
+    logging.getLogger(loggername).setState = types.MethodType(loggerSetState, logging.getLogger(loggername))
+    logging.getLogger(loggername).setLevel(logging.CRITICAL)
+
+manticore_verbosity=''
+
+def set_verbosity(setting):
+    global manticore_verbosity
+    global loggers
+
+    if isinstance(setting, (int, long)):
+        #10             20            30               40             50
+        #logging.DEBUG, logging.INFO, logging.WARNING, logging.ERROR, logging.CRITICAL
+        levels = ['',
```

Comment:
register_logger(name, letter_id=None) and let modules register themself here somehow?
Change "MANTICORE"  to "MAIN"